### ISSUE TITLE
Unable to test conditional that uses $::environment with puppet 3.8 and 4.3

### ISSUE THREAD

**Issue by italy-portugal** [rodjek/rspec-puppet#440]
Example:

``` puppet
class test {
  notify { $::environment: }
  if $::environment == 'testenv' {
      notify { 'I can test environments': }
  }
}
```

``` ruby
require 'spec_helper'
describe 'test' do
  let(:environment) { 'testenv' }
  
  # This one succeeds
  it { is_expected.to contain_notify('testenv') }

  # This one fails
  it { is_expected.to contain_notify('I can test environments') }
end
```

**Comment by trzejos** [comment]
I have had similar issues with $;;environment as well. Not exactly a fix for this bug, but you can try replacing $::environment == 'testenv' with $::environment =~ /^testenv$/ in your manifest.

**Comment by DavidS** [comment]
#453 has a minimal reproduction, and it does indeed fail on 3.8, and 4.3, but works on a range of other puppet versions.

**Comment by domcleal** [comment]
Just to be clear, the merged PR will fix it on 4.0-4.3, but it will continue to fail on 3.8.5 to .7. Apart from these few patch versions (which unfortunately are the latest 3.x), it should now work reliably.

**Comment by italy-portugal** [comment]
Thanks for the clarification!
